### PR TITLE
Add documentation for SynchronousIterator.

### PR DESCRIPTION
**Pull request by Rombur**

### CODE REVIEW

File: include/deal.II/base/synchronous_iterator.h [hunk 1]
```diff
@@ -38,6 +41,12 @@ DEAL_II_NAMESPACE_OPEN
  * The template argument of the current class shall be of type
  * <code>std_cxx11::tuple</code> with arguments equal to the iterator types.
  *
+ * The individual iterators can be accessed using 
+ * <code>std_cxx1::get<X>(synchronous_iterator.iterators)</code> where X is the
```
Comment (bangerth) [inline include/deal.II/base/synchronous_iterator.h]: std_cxx1 -> std_cxx11

File: include/deal.II/base/synchronous_iterator.h [hunk 2]
```diff
@@ -38,6 +41,12 @@ DEAL_II_NAMESPACE_OPEN
  * The template argument of the current class shall be of type
  * <code>std_cxx11::tuple</code> with arguments equal to the iterator types.
  *
+ * The individual iterators can be accessed using 
+ * <code>std_cxx1::get<X>(synchronous_iterator.iterators)</code> where X is the 
+ * number corresponding to the desired iterator.
+ *
+ * Additional documentation can be found under Files then File Members.
```
Comment (bangerth) [inline include/deal.II/base/synchronous_iterator.h]: I think this is too complicated. Are you saying that you'd like to group global functions with the class declared here? Could you use `@relates` instead?

**Comment by bangerth** [comment]
Need to re-indent.

File: include/deal.II/base/synchronous_iterator.h (hunk 2 above)
Comment (Rombur) [inline include/deal.II/base/synchronous_iterator.h]: The problem is that right now, I can't find the documentation for the global functions anywhere. By adding the ```@files``` the documentation is generated but it is hard to find.

Comment (bangerth) [inline include/deal.II/base/synchronous_iterator.h]: I know. But look at https://www.dealii.org/developer/doxygen/deal.II/classSymmetricTensor.html -- this lists all of the related global functions on the same page, which are linked via `@relates`, e.g.
```
/**
 * Compute the determinant of a tensor or rank 2. The determinant is also
 * commonly referred to as the third invariant of rank-2 tensors.
 *
 * For a one-dimensional tensor, the determinant equals the only element and
 * is therefore equivalent to the trace.
 *
 * For greater notational simplicity, there is also a <tt>third_invariant</tt>
 * function that returns the determinant of a tensor.
 *
 * @relates SymmetricTensor
 * @author Wolfgang Bangerth, 2005
 */
template <int dim, typename Number>
inline
Number determinant (const SymmetricTensor<2,dim,Number> &t)
```
I think this is a better way.

**Comment by Rombur** [comment]
Done

**Pull request merged**